cli/command/registry.go

```diff
@@ -69,9 +78,17 @@ func GetDefaultAuthConfig(cfg *configfile.ConfigFile, checkCredStore bool, serve
 			}, err
 		}
 	}
-	authCfg.ServerAddress = serverAddress
-	authCfg.IdentityToken = ""
-	return registrytypes.AuthConfig(authCfg), nil
```

Comment:
Here, we're resetting this field, but other places we either copy everything as-is, and in yet other places, we only include `Username`, `Password`, and `ServerAddress` - this needs some digging into. I suspect the `AuthConfig` struct as defined in the API contains fields that were meant to be used by the registry client, and some fields not intended to be accepted through the `X-Registry-Auth` header.